identityClient/api.go

```diff
@@ -98,7 +98,35 @@ type RegisterRealmBrokerIdentityRequest struct {
 	Identity               Identity `json:"identity"`
 }
 
-// SetRealmBackupIdentityResponse wraps values returned from a RegisterRealmBrokerIdentityRequest.
+// RegisterRealmBrokerIdentityResponse wraps values returned from a RegisterRealmBrokerIdentityRequest.
 type RegisterRealmBrokerIdentityResponse struct {
 	Identity Identity `json:"identity"`
 }
+
+// BrokerChallengeRequest wraps Identifying information for the Identity initiating the challenge.
+type BrokerChallengeRequest struct {
+	RealmName string // The name of the realm the Identity is a member of.
+	Action    string `json:"action"`   // The requested broker flow action to perform. Currently only challenge is supported.
+	Username  string `json:"username"` // The name for the Identity initiating the challenge.
+}
+
+// EmailOTP wraps a one time password provided via an email challenge.
+type EmailOTP struct {
+	EmailOTP string // The one-time password from the email challenge issued.
+}
+
+// BrokerLoginRequest proof that the Identity has completed the broker challenge
+// along with key material to encrypt the login response.
+type BrokerLoginRequest struct {
+	RealmName    string            // The name of the realm the Identity is a member of.
+	Action       string            `json:"action"`                // The requested broker flow action to perform. Currently only login is supported
+	NoteID       uuid.UUID         `json:"note_id"`               // The ID of the recovery Note the email challenge was for.
+	PublicKeys   map[string]string `json:"public_key"`            // The public key to use to encrypt the recovery note.
+	SigningKeys  map[string]string `json:"signing_key,omitempty"` // The signing key to use to verify the integrity of the recovery note.
```

Comment:
Because this is notes, this should never be empty -- any reason to `omitempty`?

Reply:
good catch, copy pasta